h/session.py

```diff
@@ -47,23 +47,39 @@ def _current_groups(request):
     This list is meant to be returned to the client in the "session" model.
 
     """
-    groups = [
-        {'name': 'Public', 'id': '__world__', 'public': True},
-    ]
+
     user = request.authenticated_user
-    if user is None:
-        return groups
-    for group in sorted(user.groups, key=_group_sort_key):
+    authority = user.authority if user else request.auth_domain
```

Comment:
If I remember the discussions correctly, the client will need to pass us the authority for which it wants the profile (and the groups).

If I'm not mistaken, an unauthenticated request on a publisher site will return the `__world__` group here, instead of the publisher group which is word-readable as well.